### PR TITLE
⭐  Kind Based Typings Split 🌟 

### PR DESCRIPTION
**Pull request by pvrameshkf**
> ### ⭐  Kind Based Typings Split 🌟 

### What ?
Based on kind generated a multiple typings file. (Lowcode & NoCode)

### Why ?
Currently we generate a typings file from lowcode.ts (which contains all the classes related apps + forms), But we are extending support for events in No Code Env (Outside of Apps) so here in the editor we have to show suggestions only for the forms.

### CODE REVIEW

File: custom.build.mjs [hunk 1]
```diff
@@ -1,27 +1,63 @@
...
-	LowcodeSDK: { name: "kf", staticDeclarations: true }
+
+const LIB_KIND = {
+  LOWCODE: "lowcode",
+  NOCODE: "nocode",
+};
+
+const BUILD_PATH_INFO = {
+  [LIB_KIND.LOWCODE]: {
+    sourceFile: "./dist/lowcode.d.ts",
```
Comment (manu2699) [inline custom.build.mjs:12]:
```suggestion
    sourcePath: "./dist/lowcode.d.ts",
```

File: custom.build.mjs [hunk 2]
```diff
@@ -31,10 +67,13 @@ function runCommand(command, callBack = null) {
 	});
 }
 
-function transfromTypings() {
-	const filePath = "./dist/lowcode.d.ts";
-	let srcFile = fs.readFileSync(filePath).toString();
-	let srcCode = tsFileStruct.parseStruct(srcFile, {}, filePath);
+function transfromTypings(params = {}) {
+	let { kind = LIB_KIND.LOWCODE } = params;
+	let { sourceFile, destinationFile } = BUILD_PATH_INFO[kind];
```
Comment (manu2699) [inline custom.build.mjs:72]:
```suggestion
	let { sourcePath, destinationPath } = BUILD_PATH_INFO[kind];
```

Comment (pvrameshkf) [reply custom.build.mjs]: done

File: custom.build.mjs [hunk 3]
```diff
@@ -1,27 +1,63 @@
...
-	Client: { name: "Client" },
-	Component: { name: "Component" },
-	Form: { name: "Form" },
-	Table: { name: "Table" },
-	TableForm: { name: "TableForm" },
-	Popup: { name: "Popup" },
-	Formatter: { name: "Formatter" },
-	LowcodeSDK: { name: "kf", staticDeclarations: true }
+
+const LIB_KIND = {
```
Comment (satheesh66) [inline custom.build.mjs:5]: LIBRARY_KIND

Comment (pvrameshkf) [reply custom.build.mjs]: Changed the name into SDK_KIND since its more readable than LIBRARY_KIND.

File: custom.build.mjs [hunk 4]
```diff
@@ -1,27 +1,63 @@
...
+		{ kind: SDK_KIND.NOCODE },
+		transfromTypings
+  	);
 });
 
-function runCommand(command, callBack = null) {
+function runCommand(command, params = {}, callBack = null) {
 	const execCmd = exec(command, () => {
-		callBack && callBack();
+		callBack && callBack(params);
```
Comment (saravanan10393) [inline custom.build.mjs:60]: assign default arg as function `arg=()=> {}` and remove the `callBack &&` condition

**Pull request merged**